modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -228,17 +224,20 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
     SequenceView(seq.id, seq.metadata, st, engineSteps(seq), None)
   }
 
-  private def refreshSequenceList(q: EventQueue): Engine.State => Task[Unit] = (st: Engine.State) => {
+  private def refreshSequenceList(q: EventQueue): Engine.State => Task[Option[Process[Task, Event]]] = (st: Engine.State) => {
 
     val seqexecList = st.sequences.keys.toSeq.map(v => new SPObservationID(v))
 
-    def loads(odbList: Seq[SPObservationID]): Seq[Task[SeqexecFailure \/ Unit]] = odbList.diff(seqexecList).map(id => load(q, id))
+    def loads(odbList: Seq[SPObservationID]): Task[List[Event]] = odbList.diff(seqexecList).toList.map(id => loadEvents(id)).sequenceU.map(_.flatten).run.map(_.valueOr( r => List(Event.logMsg(SeqexecFailure.explain(r)))))
 
-    def unloads(odbList: Seq[SPObservationID]): Seq[Task[SeqexecFailure \/ Unit]] = seqexecList.diff(odbList).map(id => unload(q, id))
+    def unloads(odbList: Seq[SPObservationID]): Seq[Event] = seqexecList.diff(odbList).map(id => unload(id))
```

Comment:
`loads` reads data from an external source, that is why each load is in a `Task`. It can also fail, therefore the Task output is in a `\/`.
The case of `unloads` is different. The computation does not have side effects, and it cannot fail. Therefore, you can just output the result as it is.